projects/plugins/jetpack/extensions/shared/components/product-management-controls/toolbar-control.js

```diff
@@ -70,11 +79,33 @@ function Product( { onClose, product } ) {
 
 function NewProduct( { onClose } ) {
 	const { clientId, productType } = useProductManagementContext();
+	const siteSlug = useSelect( select => select( membershipProductsStore ).getSiteSlug() );
 	const openBlockSidebar = useOpenBlockSidebar( clientId );
 
+	const isPublishOpen = useSelect(
+		select => select( 'core/edit-post' ).isPublishSidebarOpened(),
+		[]
+	);
```

Comment:
Right now we have a "component" that has the responsibility of opening the block sidebar, you can find the definition of that component [here](https://github.com/Automattic/jetpack/blob/6558004ac2fe4d1f23885995f485b33d0b14cc14/projects/plugins/jetpack/extensions/shared/components/product-management-controls/use-open-block-sidebar.js). 

The problem domain of the use-open-block-sidebar is to perform all necessary operations to open the sidebar for a given blockId. I think we should include this code in that component instead of having it in the toolbar-control component (whose problem domain is just to show the product management controls).

If we don't include this code in the use-open-block-sidebar component, every other new component that we create will need to re-implement a way to hide the publishing sidebar before opening the sidebar which will lead to increased maintenance costs in the long run (and multiple sources of truth for the same operation).

I've created a diff with how it would look, feel free to use it and/or modify it. It also includes the solution to my other comment.

```
diff --git a/projects/plugins/jetpack/extensions/shared/components/product-management-controls/toolbar-control.js b/projects/plugins/jetpack/extensions/shared/components/product-management-controls/toolbar-control.js
index 6dd7531004..3918aab0f8 100644
--- a/projects/plugins/jetpack/extensions/shared/components/product-management-controls/toolbar-control.js
+++ b/projects/plugins/jetpack/extensions/shared/components/product-management-controls/toolbar-control.js
@@ -19,15 +19,7 @@ import { useProductManagementContext } from './context';
 import useOpenBlockSidebar from './use-open-block-sidebar';
 import { getMessageByProductType } from './utils';
 import { store as membershipProductsStore } from '../../../store/membership-products';
-
-/**
- * Check if it's in the context of the customizer.
- *
- * @returns {boolean} if we are in the context of Customizer.
- */
-function isInCustomizer() {
-	return 'function' === typeof window?.wp?.customize;
-}
+import { CUSTOMIZER_EDITOR, getEditorType } from '../../get-editor-type';
 
 function getProductDescription( product ) {
 	const { currency, interval, price } = product;
@@ -82,12 +74,7 @@ function NewProduct( { onClose } ) {
 	const siteSlug = useSelect( select => select( membershipProductsStore ).getSiteSlug() );
 	const openBlockSidebar = useOpenBlockSidebar( clientId );
 
-	const isPublishOpen = useSelect(
-		select => select( 'core/edit-post' ).isPublishSidebarOpened(),
-		[]
-	);
-
-	if ( isInCustomizer() ) {
+	if ( CUSTOMIZER_EDITOR === getEditorType() ) {
 		return (
 			<MenuItem>
 				{ siteSlug && (
@@ -103,9 +90,6 @@ function NewProduct( { onClose } ) {
 		event.preventDefault();
 		openBlockSidebar();
 
-		// We need to close the publish sidebar when the user tries to add a new subscription otherwise the block panel is not visible.
-		isPublishOpen && dispatch( 'core/edit-post' ).closePublishSidebar();
-
 		setTimeout( () => {
 			const input = document.getElementById( 'new-product-title' );
 			if ( input !== null ) {
diff --git a/projects/plugins/jetpack/extensions/shared/components/product-management-controls/use-open-block-sidebar.js b/projects/plugins/jetpack/extensions/shared/components/product-management-controls/use-open-block-sidebar.js
index dd72de8990..3933361b4b 100644
--- a/projects/plugins/jetpack/extensions/shared/components/product-management-controls/use-open-block-sidebar.js
+++ b/projects/plugins/jetpack/extensions/shared/components/product-management-controls/use-open-block-sidebar.js
@@ -2,9 +2,13 @@
  * WordPress dependencies
  */
 import { store as blockEditorStore } from '@wordpress/block-editor';
-import { useDispatch } from '@wordpress/data';
+import { dispatch, useDispatch, select } from '@wordpress/data';
 import { getEditorType, SITE_EDITOR, WIDGET_EDITOR } from '../../get-editor-type';
 
+const closePublishSidebarIfNeeded = () =>
+	select( 'core/edit-post' ).isPublishSidebarOpened() &&
+	dispatch( 'core/edit-post' ).closePublishSidebar();
+
 export default function useOpenBlockSidebar( clientId ) {
 	const editorType = getEditorType();
 	const { selectBlock } = useDispatch( blockEditorStore );
@@ -21,6 +25,7 @@ export default function useOpenBlockSidebar( clientId ) {
 			case WIDGET_EDITOR:
 				return enableComplementaryArea( 'core/edit-widgets', 'edit-widgets/block-inspector' );
 			default:
+				closePublishSidebarIfNeeded();
 				enableComplementaryArea( 'core/edit-post', 'edit-post/block' );
 		}
 	};

```

Reply:
Good idea! I'll make the changes. :)